### PR TITLE
Workarround for #31152 and #34509

### PR DESCRIPTION
**Pull request by Dominic12**
This is a workarround for #34509 #31152.

### CODE REVIEW

**Comment by Dominic12** [comment]
Added CLA

**Comment by Dominic12** [comment]
I signed it!

**Comment by Dominic12** [comment]
I signed it!

**Comment by Dominic12** [comment]
Signed CLA

**Comment by Dominic12** [comment]
I signed It!

File: lib/ui/plugins/callback_cache.cc [hunk 1]
```diff
@@ -126,14 +126,14 @@ void DartCallbackCache::LoadCacheFromDisk() {
   }
   std::string cache_contents{std::istreambuf_iterator<char>(input),
                              std::istreambuf_iterator<char>()};
-  Document d;
-  d.Parse(cache_contents.c_str());
-  if (d.HasParseError() || !d.IsArray()) {
+  Document document;
```
Comment (cyanglaz) [inline lib/ui/plugins/callback_cache.cc]: let's not have these rename in this PR. seems not related.

**Comment by Dominic12** [comment]
Jup, i‘ll remove the refactoring when i have time. And yes, this will reintroduce a Memoryleak, which has been fixed. If you want it to be a real fix instead of a Workaround feel free to close this PR and i will take the time to create a real fix. But keep in mind, that in my opinion the crash is more severe than a memoryleak.

File: lib/ui/plugins/callback_cache.cc (hunk 1 above)
Comment (Dominic12) [reply lib/ui/plugins/callback_cache.cc]: Jup, i‘ll remove the refactoring when i have time. And yes, this will reintroduce a Memoryleak, which has been fixed. If you want it to be a real fix instead of a Workaround feel free to close this PR and i will take the time to create a real fix. But in my opinion the crash is more severe than a memoryleak.

**Comment by Dominic12** [comment]
Okay, i'll investigate further. Feel free to close this issue.

**Comment by Dominic12** [comment]
So after consulting the documentation from apple:  [(Documentation)](https://developer.apple.com/documentation/opengles/eaglcontext?language=objc) the function setCurrentContext seems to be deprecated anyway.

**Comment by Dominic12** [comment]
Screenshot 2019-06-19 at 20 11 46

Flutter has some memory leaks with my PR, but i'm not sure if they are related to this PR.

**Comment by cyanglaz** [comment]
We are going with another temp solution which won't reintroduce the leak. https://github.com/flutter/engine/pull/9377
Closing this PR now.

File: lib/ui/plugins/callback_cache.cc (hunk 1 above)
Comment (darkmarthur) [reply lib/ui/plugins/callback_cache.cc]: I think the same
